207 past events page and events list styles

suejinkim20 opened the pull request.

mbwatson commented on src/templates/events/event-template.js:
```diff
@@ -116,8 +117,26 @@ export default ({ data, pageContext }) => {
 
+  const todaysDate = new Date();
 
+  const dateString = `${todaysDate.getFullYear()}-${
```
you can actually compare two Date objects directly with `>`, `<`, `<=` or `>=`, so there's no need to construct date strings to compare.

mbwatson commented on src/templates/events/event-template.js:
```diff
@@ -116,8 +117,26 @@ export default ({ data, pageContext }) => {
+    parsedEventDate.getDate() < 10 ? "0" : ""
+  }${parsedEventDate.getDate()}`
+
+  const past = dateString > eventDate ? true : false
```
the expression `dateString > eventData` returns `true` or `false` already, so there's no need to check if that expression is `true` to determine whether to return `true` or `false`; you can simply return the expression.

mbwatson commented on src/templates/events/upcoming-events-template.js:
```diff
@@ -28,49 +28,47 @@ const EventsList = ({ title, events }) => {
-              {/* {!fabricHosted && url && (
-                <Meta
+            <Fragment key={title}>
+              <Subsubheading 
```
can you flip these headings? this order results in out-of-order H3 and H4 tags.

mbwatson commented on src/templates/events/upcoming-events-template.js:
```diff
@@ -28,49 +28,47 @@ const EventsList = ({ title, events }) => {
+                {display_date ? display_date : date}
+              </Subsubheading>
+              <Subheading 
+                style={{
```
are these styles that belong at a higher level so headings can have a consistent look? these styles don't seem very different that the component's default styles, so perhaps it would be more appropriate to change the base component's style instead of having these one-off, differently styled headings. we want to keep long-term maintenance in mind here.

mbwatson commented on src/templates/events/upcoming-events-template.js:
```diff
@@ -28,49 +28,47 @@ const EventsList = ({ title, events }) => {
-                    justifyContent: "flex-start",
-                    alignItems: "flex-start",
-                    flexDirection: isCompact ? "column" : "row",
+                    textUnderlineOffset: "0.3rem",
```
please don't do this. style the base component. text links should have consistent styling.

mbwatson commented on src/templates/events/upcoming-events-template.js:
```diff
@@ -28,49 +28,47 @@ const EventsList = ({ title, events }) => {
+                >{title}</Link>
+              </Subheading>
               <Paragraph
                 style={{
```
similar comment here as the above ones. styling consistency.

suejinkim20 commented:
@mbwatson since there's a lot going on with this PR, here's a summary of changes:
- cleaned up url paths for all events
- removed unneeded variables from md files
- cleaned up display date formats for events so they are uniform
- added a "forum_post" to link session materials when an event has passed
- updated styles on the aggregate event list page
- added a past event alert box component to let users know that an event has past and if there are session materials to view
- tweaked typography components to add a few customized props (for margin and alignment)

mbwatson reviewed (approved):
:+1: looks good!
